parsecs/src/test/scala/ParserTests.scala

```diff
@@ -1,13 +1,91 @@
 package com.jpmc.parsecs
 
-import org.scalatest.funsuite.AnyFunSuite
-import org.scalatest.matchers.should.Matchers
-import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
-
-class ParserTests extends AnyFunSuite with ScalaCheckPropertyChecks with Matchers {
-  test("Some test") {
-    forAll { str: String =>
-      str should be(str)
+import com.jpmc.parsecs.Parser.{string, whitespace}
+import com.jpmc.parsecs.Result.Success
+import org.scalatest.freespec.AnyFreeSpec
+
+
+class ParserTests extends AnyFreeSpec {
+  "Parse string" in {
+    val str = "Hello"
+    val result = Parser.string("Hello").parse(str)
+    val expected = Result.Success("Hello", "")
+    assert(result == expected)
+  }
+  "Parse string partial" in {
+    val str = "Hello"
+    val result = Parser.string("Hell").parse(str)
+    val expected = Result.Success("Hell", "o")
+    assert(result == expected)
+  }
+  "Parse Char" in {
+    val c = "c".toCharArray.head
+    val result = Parser.char(c).parse("c")
+    val expected = Result.Success(c, "")
+    assert(result == expected)
+  }
+  "Parse Int" in {
+    val num = "123"
+    val result = Parser.number.parse(num)
+    val expected = Result.Success(123, "")
+    assert(result == expected)
+  }
+  "Parse Boolean" - {
+    "true" in {
+      val t = "true"
+      val result = Parser.boolean.parse(t)
+      val expected = Result.Success(true, "")
+      assert(result == expected)
+    }
+    "false" in {
+      val f = "false"
+      val result = Parser.boolean.parse(f)
+      val expected = Result.Success(false, "")
+      assert(result == expected)
+    }
+    "faalse" in {
+      val f = "faalse"
+      val result = Parser.boolean.parse(f)
+      val expected = Result.Failure("Unable to parse input: alse")
+      assert(result == expected)
+    }
+  }
+  "Parse Name" - {
+    "Success" in {
+      val firstName  = "Monica"
+      val lastName  = "McGuigan"
+      def parseName(firstName: String, secondName: String): Parser[(String, String)] = (string(firstName) <* whitespace.rep0) ~ string(secondName)
+
+      val result = parseName(firstName, lastName).parse("MonicaMcGuigan")
+      val expected = Success(("Monica", "McGuigan"), "")
+      assert(result == expected)
+    }
+  }
+  "Parse opt" in {
+    val str = "string"
+    val res = Parser.string("string").opt.parse(str)
+    val expected = Success(Some(str), "")
+    assert(res == expected)
+  }
+
+  "List" - {
```

Comment:
I like this test here, it’s a pretty important edge case

Reply:
thanks 😇